src/compiler/checker.ts

```diff
@@ -4662,6 +4662,9 @@ namespace ts {
                 // type arguments in the same manner as a type reference to get the same error reporting experience.
                 baseType = getTypeFromClassOrInterfaceReference(baseTypeNode, baseConstructorType.symbol);
             }
+            else if (baseConstructorType.flags & TypeFlags.Any) {
+                baseType = anyType;

```

Comment:
There's no good reason; in fact it reduces test churn to return baseConstructorType for tests that extend undefined variables, which I think get `unknownType` instead of `anyType`, meaning that you get slightly more errors.

I'll change it to baseConstructorType.